src/__tests__/unit/controllers/hooks/router-store/test.tsx

```diff
@@ -0,0 +1,342 @@
+
+            return <div>I am a subscriber</div>;
+          }}
+        </MockComponent>
+      </Router>
+    );
+  });
+
+  it('should return undefined for non-existent params', () => {
+    mount(
+      <Router routes={mockRoutes} history={history}>
+        <MockComponent>
+          {() => {
+            const [param] = useQueryParam('iamnotaqueryparam');
+            expect(param).toEqual(undefined);
+
+            return <div>I am a subscriber</div>;
+          }}
+        </MockComponent>
+      </Router>
+    );
+  });
+
+  it('should return undefined for non-existent params and update the URL when set for the first time', async () => {
+    const mockPath = mockLocation.pathname;
+    let qpVal: string | undefined, qpUpdateFn: (qp: string) => void;
+
+    mount(
+      <Router routes={mockRoutes} history={history}>
+        <MockComponent>
+          {() => {
+            const [param, setParam] = useQueryParam('newqueryparam');
+            qpVal = param;
+            qpUpdateFn = setParam;
+
+            return <div>I am a subscriber</div>;
+          }}
+        </MockComponent>
+      </Router>
+    );
+    expect(qpVal).toEqual(undefined);
+    act(() => qpUpdateFn('val'));
+    await nextTick();
+
+    expect(historyPushSpy).toBeCalledWith(
+      `${mockPath}?foo=hello&bar=world&newqueryparam=val#hash`
+    );
+  });
+
+  it('should update URL with new param value', async () => {
+    const mockPath = mockLocation.pathname;
+    let qpVal: string | undefined, qpUpdateFn: (qp: string) => void;
+
+    mount(
+      <Router routes={mockRoutes} history={history}>
+        <MockComponent>
+          {() => {
+            const [param, setParam] = useQueryParam('foo');
+            qpVal = param;
+            qpUpdateFn = setParam;
+
+            return <div>I am a subscriber</div>;
+          }}
+        </MockComponent>
+      </Router>
+    );
+
+    expect(qpVal).toEqual('hello');
+    act(() => qpUpdateFn('newVal'));
+    await nextTick();
+
+    expect(historyPushSpy).toBeCalledWith(
+      `${mockPath}?foo=newVal&bar=world#hash`
+    );
+  });
+
+  it('should remove param from URL when set to null', async () => {
+    const mockPath = mockLocation.pathname;
+    let qpVal: string | undefined, qpUpdateFn: (qp: string | null) => void;
+
+    mount(
+      <Router routes={mockRoutes} history={history}>
+        <MockComponent>
+          {() => {
+            const [param, setParam] = useQueryParam('foo');
+            qpVal = param;
+            qpUpdateFn = setParam;
+
+            return <div>I am a subscriber</div>;
+          }}
+        </MockComponent>
+      </Router>
+    );
+
+    expect(qpVal).toEqual('hello');
+
+    act(() => qpUpdateFn(null));
+
+    await nextTick();
+
+    expect(historyPushSpy).toBeCalledWith(`${mockPath}?bar=world#hash`);
+  });
+
+  it('should only re-render components hooked to a specific param', async () => {
+    let fooVal: string | undefined,
+      barVal: string | undefined,
```

Comment:
Same as other comment before, let's use a `let` per declaration

Reply:
Changed to use a `let` per declaration